hydromt_wflow/wflow.py

```diff
@@ -1584,6 +1585,189 @@ def setup_temp_pet_forcing(
+        return_period: Optional[list] = [2,3,5,10,15,20,25,50,60,100],
+        Imax: Optional[float] = 2.0,
+        start_hydro_year: Optional[str] = "Sep",
+        start_field_capacity: Optional[str] = "Apr",
+        LAI: Optional[bool] = False,
+        rooting_depth: Optional[bool] = True,
+        correct_cc_deficit: Optional[bool] = False,
+        time_tuple: Optional[tuple] = None,
+        time_tuple_fut: Optional[tuple] = None,
+        missing_days_threshold: Optional[int] = 330,
+        update_toml_rootingdepth: Optional[str] = "RootingDepth_obs_20",
+        **kwargs,
+    ) -> None:
+        """
+        This component sets up the RootingDepth by estimating the catchment-scale 
+        root-zone storage capacity from observed hydroclimatic data 
+        (and optionally also for climate change historical and future periods). 
+        This presents an alternative approach to determine the RootingDepth 
+        based on hydroclimatic data instead of through a look-up table relating 
+        land use to rooting depth (as usually done for the wflow_sbm model). 
+        The method is based on the estimation of maximum annual storage deficits 
+        based on precipitation and estimated actual evaporation time series, 
+        which in turn are estimated from observed streamflow data and 
+        long-term precipitation and potential evap. data, as explained in 
+        Bouaziz et al. (2022).
+        The main assumption is that vegetation adapts its rootzone storage capacity 
+        to overcome dry spells with a certain return period (typically 20 years for forest ecosystems). 
+        In response to a changing climtate, 
+        it is likely that vegetation also adapts its rootzone storage capacity, 
+        thereby changing model parameters for future conditions.
+        This method also allows to estimate the change in rootzone storage capacity 
+        in response to a changing climate. 
+
+        As the method requires precipitation and potential evaporation timeseries, it may be useful to run this method as an update step in the setting-up of the hydrological model, once the forcing files have already been derived. 
+        In addition the setup_soilmaps method is also required if RootingDepth are to be calculated. 
+
+        References
+        ----------
+        Bouaziz, L. J. E., Aalbers, E. E., Weerts, A. H., Hegnauer, M., Buiteveld, 
+        H., Lammersen, R., Stam, J., Sprokkereef, E., Savenije, H. H. G. and 
+        Hrachowitz, M. (2022). Ecosystem adaptation to climate change: the 
+        sensitivity of hydrological predictions to time-dynamic model parameters, 
+        Hydrology and Earth System Sciences, 26(5), 1295-1318. DOI: 
+        10.5194/hess-26-1295-2022.
+
+        Adds model layer:
+
+        * **rootzone_storage_{forcing}_{RP}** map: rootzone storage capacity [mm of water] estimated from hydroclimatic data {forcing: obs, cc_hist or cc_fut} for different return periods RP.
```

Comment:
These maps, are they ever used by wflow_sbm? Else I'm starting to wonder if we should not add them to auxmaps and starts using that object in hydromt_wflow? I have that as well for the different Ksat per layer in staticmaps too. In hydromt we produce a lot of extra maps that are actually not used and I think confuse users when we give them an example wflow model. What do you think?

Else it's something I can look into maybe in the GridModel PR where we also switch from staticmaps to grid.

Reply:
good to have a look later at the auxmaps options! for now I replaced the rootzone_depth flag with a rootzone_storage flag (set as default to False) to only store the RootingDepth maps and no longer the rootzone_storage maps, which are not used by sbm. so this will reduce the amount of new maps in the staticmaps.